Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -352,9 +331,8 @@ bool loadCanceled
         document, statusPublisher, NotificationPublisher);
     }
 
-    private async Task NotifyStatusAsync(TextDocumentItem item, IObservable<DafnyDocument> documents, CancellationToken cancellationToken) {
-      var finalDocument = await documents.ToTask(cancellationToken);
-      var errorCount = finalDocument.ImplementationIdToView.Values.Sum(r => r.Diagnostics.Count(d => d.Severity == DiagnosticSeverity.Error));
+    private void NotifyStatusAsync(TextDocumentItem item, DafnyDocument document, CancellationToken cancellationToken) {
+      var errorCount = document.ImplementationIdToView.Values.Sum(r => r.Diagnostics.Count(d => d.Severity == DiagnosticSeverity.Error));
```

Comment:
How safe is it to iterate on a concurrent dictionary like this without locking?

Reply:
I think `.Values` makes a copy while keeping a lock. Also, this code is run after all verification tasks are completed and the dictionary is no longer updated.